src/Microsoft.AspNet.SignalR.ServiceBus3/ServiceBusConnection.cs

```diff
@@ -0,0 +1,328 @@
+                }
+
+                // Create a receiver to get messages
+                string subscriptionEntityPath = SubscriptionClient.FormatSubscriptionPath(topicName, subscriptionName);
+                MessageReceiver receiver = _factory.CreateMessageReceiver(subscriptionEntityPath, ReceiveMode.ReceiveAndDelete);
+
+                _trace.TraceInformation("Creation of a message receive for subscription entity path {0} in the service bus completed successfully.", subscriptionEntityPath);
+
+                connectionContext.SetSubscriptionContext(new SubscriptionContext(topicName, subscriptionName, receiver), topicIndex);
+
+                var receiverContext = new ReceiverContext(topicIndex, receiver, connectionContext);
+
+                ProcessMessages(receiverContext);
+
+                // Open the stream
+                connectionContext.OpenStream(topicIndex);
+            }
+        }
+
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "We retry to create the topics on exceptions")]
+        private void Retry(Action action)
+        {
+            string errorMessage = "Failed to create service bus subscription or topic : {0}";
+            while (true)
+            {
+                try
+                {
+                    action();
+                    break;
+                }
+                catch (UnauthorizedAccessException ex)
+                {
+                    _trace.TraceError(errorMessage, ex.Message);
+                    break;
+                }
+                catch (MessagingException ex)
+                {
+                    _trace.TraceError(errorMessage, ex.Message);
+                    if (ex.IsTransient)
+                    {
+                        Thread.Sleep(RetryDelay);
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _trace.TraceError(errorMessage, ex.Message);
+                    Thread.Sleep(RetryDelay);
+                }
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                // Close the factory
+                if (_factory != null)
+                {
+                    _factory.Close();
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Exceptions are handled through the error handler callback")]
+        private void ProcessMessages(ReceiverContext receiverContext)
+        {
+            receiverContext.Receiver.ReceiveBatchAsync(ReceiverContext.ReceiveBatchSize, receiverContext.ReceiveTimeout)
```

Comment:
This is a random background thread. You can also just task.run or use this beauty https://github.com/aspnet/SignalR/blob/eafbe741609981cdcf6b6f28d45cb2f50126750d/src/Microsoft.AspNetCore.Sockets/Internal/AwaitableThreadPool.cs. Usage here https://github.com/aspnet/SignalR/blob/eafbe741609981cdcf6b6f28d45cb2f50126750d/src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs#L291